File: backend/finan-exp-backend/Customers/Models/Customer.cs
using Customers.Enums;
using Shared.Classes;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Customers.Models
{
  [Table("Customers")]
  public class Customer: FullEntity
  {
    [Required]
    public string Name { get; set; }

    [Required]
    public CustomersTypeOptions Type { get; set; }

    [Required]
    public decimal InitialBalance { get; set; }

    [Required]
    public decimal ActualBalance { get; set; }

    [Required]
    public decimal TransferValue { get; set; }

    [Required]
    public bool IsArchived { get; set; }
  }
}

File: src/app/core/models/customers/customer.ts
import { customerTypesOptions } from './../../../shared/support/enums/customer-types-options';
import { FullEntity } from 'dist/finan-exp-common';

export abstract class Customer extends FullEntity {
  public name!: string;

  public type!: customerTypesOptions;

  public initialBalance!: number;

  public actualBalance!: number;

  public transferValue!: number;

  public isArchived!: boolean;
}
